receiver/splunksearchapireceiver/client.go

```diff
@@ -59,7 +60,7 @@ func newSplunkSearchAPIClient(ctx context.Context, settings component.TelemetryS
 func (c defaultSplunkSearchAPIClient) CreateSearchJob(search string) (CreateJobResponse, error) {
 	endpoint := fmt.Sprintf("%s/services/search/jobs", c.endpoint)
 
-	reqBody := fmt.Sprintf(`search=%s`, search)
+	reqBody := fmt.Sprintf(`search=%s`, url.QueryEscape(search))
```

Comment:
Are we sure we want to be url.QueryEscaping in this circumstance? I thought that was for query parameters; Do you have a case that caused this to be necessary?

Reply:
@schmikei from the example in the [API docs](https://docs.splunk.com/Documentation/Splunk/9.3.1/RESTREF/RESTsearch#search.2Fjobs):
```
curl -k -u admin:pass https://localhost:8089/servicesNS/admin/search/search/jobs --data-urlencode search="search index=_internal source=*/metrics.log"
```
That's the only example that uses --data-urlencode, but since we'll be adding the time fields to every search, there's not a case where we wouldn't need it to get those right (unless there's some other way)

Reply:
Looked into this a bit further and yep it seems like this is the correct approach when using the Splunk search REST API https://community.splunk.com/t5/Splunk-Enterprise/rest-api-search-not-working-when-using-cs-uri-stem-cs-uri-query/m-p/532759